packages/react-server/core/context/Navigator.js

```diff
@@ -203,7 +204,11 @@ class Navigator extends EventEmitter {
 			// TODO: I think that 3xx/4xx/5xx shouldn't be considered "errors" in navigateDone, but that's
 			// how the code is structured right now, and I'm changing too many things at once at the moment. -sra.
 			if (handleRouteResult.code && ((handleRouteResult.code / 100)|0) !== 2) {
-				this.emit("navigateDone", {status: handleRouteResult.code, redirectUrl: handleRouteResult.location}, page, request.getUrl(), type);
+
+				let error = new HttpError("Page returned code " + handleRouteResult.code, {code: handleRouteResult.code});
+				error.redirectUrl = handleRouteResult.location;
+				page.setRedirectUrl(handleRouteResult.location);
+				this.emit("navigateDone", error, page, request.getUrl(), type);
```

Comment:
`HttpError` should have members `status` and `redirectUrl` same as the object that was being returned before, but this should be tested further.